gc_registry/certificate/schemas.py

```diff
@@ -1,12 +1,173 @@
+    )
+
+    ### Bundle Characteristics ###
+    energy_carrier: str = Field(
+        description="The form of energy that the GC Bundle represents, for example: Electricity, Hydrogen, Ammonia. In the current version of the standard (v2), this field is always Electricity.",
+    )
+    energy_source: str = Field(
+        description="The fuel type used to generate the energy represented by the GC Bundle, for example: Solar, Wind, Biomass, Nuclear, Coal, Gas, Oil, Hydro.",
+    )
+    face_value: int = Field(
+        description="States the quantity of energy in Watt-hours (Wh) represented by each Granular Certificate within this GC Bundle.",
+    )
+    issuance_post_energy_carrier_conversion: bool = Field(
+        description="Indicate whether this GC Bundle have been issued following an energy conversion event, for example in a power to hydrogen facility.",
+    )
+
+    ### Production Device Characteristics ###
+    device_id: int = Field(
+        foreign_key="device.id",
+        description="Each GC Bundle is associated with a single production Device.",
+    )
+
+    ### Temporal Characteristics ###
+    production_starting_interval: datetime.datetime = Field(
+        description="""The datetime in UTC format indicating the start of the relevant production period.
+                        GC Bundles shall be issued over a maximum production period of one hour,
+                        under the assumption that the certificates represent an even distribution of power generation within that period.""",
+    )
+    production_ending_interval: datetime.datetime = Field(
+        description="The datetime in UTC format indicating the end of the relevant production period.",
+    )
+    issuance_datestamp: datetime.datetime = Field(
+        description="The date in UTC format (YYYY-MM-DD) indicating the date on which the Issuing Body delivered the GC Bundle to the production Device's registered Account.",
+    )
+    expiry_datestamp: datetime.datetime = Field(
+        description="The date in UTC format (YYYY-MM-DD) indicating the point at which the GC Bundle will be rendered invalid if they have not been cancelled. This expiry period can vary across Domains.",
+    )
+
+    ### Storage Characteristics ###
+    is_storage: int = Field(
+        description="Indicates whether the Device ID is associated with a storage Device.",
+    )
+    sdr_allocation_id: int | None = Field(
+        default=None,
+        description="The unique ID of the Storage Discharge Record that has been allocated to this GC Bundle.",
+        foreign_key="storagedischargerecord.sdr_allocation_id",
+    )
+    storage_efficiency_factor: float | None = Field(
+        default=None,
+        description="The efficiency factor of the storage Device that has discharged the energy represented by this GC Bundle.",
+    )
+
+
+class GranularCertificateBundleCreate(GranularCertificateBundleBase):
+    pass
+
+
+class GranularCertificateRegistryBase(BaseModel):
```

Comment:
This is likely me being thick, but at what point downstream would these be joined to a GC bundle? There's no field that you could join on, and some of these attrs (e.g. `legal_status`, `issuance_purpose`) will vary by certificate rather than stay static across the registry

Reply:
The GCBBase has foreign key `registry_id`, but not sure if it is worth having attributes in a table that are completely static per registry?

`legal_status` we have described as follows: `"May contain pertinent information on the Issuing Authority, where relevant."`. Assumed it was a static legal status of the issuer, but can check.

Agree `issuance_purpose` might not belong on a registry table if we continue with this approach.

`support_received` looks like it belongs on the device